partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -621,71 +625,29 @@ internal class PlanTyper(
         private fun toRexCall(
             match: FnMatch.Ok<FunctionSignature.Scalar>,
             args: List<Rex>,
-            isNotMissable: Boolean,
         ): Rex {
             // Found a match!
             val newFn = fnResolved(match.signature)
             val newArgs = rewriteFnArgs(match.mapping, args)
-            val returns = newFn.signature.returns
 
-            // 7.1 All functions return MISSING when one of their inputs is MISSING (except `=`)
-            newArgs.forEach {
-                if (it.type == MissingType && !isNotMissable) {
-                    handleAlwaysMissing()
-                    return rex(MISSING, rexOpCallStatic(newFn, newArgs))
-                }
+            // Check literal missing inputs
+            val argAlwaysMissing = args.any {
+                val op = it.op as? Rex.Op.Lit ?: return@any false
+                op.value.type == PartiQLValueType.MISSING
```

Comment:
similar to other comment, should check if something is a missing literal by checking if it's the literal missing value?

```suggestion
                op.value == missingValue()
```

Reply:
The suggestion is a method invocation resulting in a whole new Java class instance and uses the equals method to check the Java class and properties. The original is a check on an enum (integer). Also, this is similar to the reasoning behind PQLValue -- we should rely on our own type constructs rather than Java's.

Reply:
I should have linked the comment for clarity -- https://github.com/partiql/partiql-lang-kotlin/pull/1463#discussion_r1604083403. We're calling `PartiQLValueType.MISSING` here but since it's deprecated in this PR and will be removed in a future release, I'm thinking we should not call the enum variant. Here's a version not calling the function invocation:

```suggestion
                op.value is MissingValue
```